Source/WebCore/css/StyleColor.h

```diff
@@ -31,31 +30,119 @@
 
 #pragma once
 
-#include "CSSValueKeywords.h"
 #include "Color.h"
+#include "ColorInterpolationMethod.h"
+#include "CSSValueKeywords.h"
+#include <vector>
 #include <wtf/OptionSet.h>
 
 namespace WebCore {
 
+class CSSPrimitiveValue;
+
 enum class StyleColorOptions : uint8_t {
     ForVisitedLink = 1 << 0,
     UseSystemAppearance = 1 << 1,
     UseDarkAppearance = 1 << 2,
     UseElevatedUserInterfaceLevel = 1 << 3
 };
 
-struct StyleColor {
+class StyleColor;
+
+struct CurrentColor {
+    bool operator==(CurrentColor const&) const = default;
+    static StyleColor make();
+};
+
+struct SystemColor {
+    bool operator==(SystemColor const&) const = default;
+};
+struct ColorMixComponent;
+
+struct ColorContext {
+    std::optional<Color> currentColor = std::nullopt;
+};
+
+struct ColorMixPercentages {
+    double p1 = 0;
+    double p2 = 0;
+    std::optional<double> alphaMultiplier = std::nullopt;
+    bool operator==(ColorMixPercentages const&) const = default;
+    friend WTF::TextStream& operator<<(WTF::TextStream&, ColorMixPercentages const&);
+};
+
+struct ColorMix {
+    ColorMix(ColorInterpolationMethod, ColorMixComponent const&, ColorMixComponent const&, ColorMixPercentages const&);
+    bool operator==(ColorMix const&) const = default;
+    Color mixComponents(ColorContext const&) const;
+    friend WTF::TextStream& operator<<(WTF::TextStream&, ColorMix const&);
+private:
+    ColorInterpolationMethod m_colorInterpolationMethod;
+    std::vector<ColorMixComponent> m_components;
+    ColorMixPercentages m_percentages;
+};
+
+using ColorKind = std::variant<Color, CurrentColor, SystemColor, ColorMix>;
+using CSSColorKind = std::variant<CSSPrimitiveValue, ColorMix>;
+
+class StyleColor {
+public:
     static Color colorFromKeyword(CSSValueID, OptionSet<StyleColorOptions>);
+    static Color colorFromAbsoluteKeyword(CSSValueID);
     static bool isAbsoluteColorKeyword(CSSValueID);
+    static bool isCurrentColorKeyword(CSSValueID);
+    static bool isCurrentColor(CSSPrimitiveValue const&);
     WEBCORE_EXPORT static bool isSystemColorKeyword(CSSValueID);
 
     enum class CSSColorType : uint8_t {
         Absolute = 1 << 0,
         Current = 1 << 1,
         System = 1 << 2,
     };
+
     // https://drafts.csswg.org/css-color-4/#typedef-color
     static bool isColorKeyword(CSSValueID, OptionSet<CSSColorType> = { CSSColorType::Absolute, CSSColorType::Current, CSSColorType::System });
+
+    StyleColor() = default;
+    StyleColor(Color const&);
```

Comment:
You have this in quite a few places, but WebKit style is to write the `const` first, so `const Color&`

Reply:
@mdubet This doesn't seem fixed